### PR TITLE
Rename variables and methods

### PR DESCRIPTION
**Pull request by syershov**

### CODE REVIEW

File: generator/transit_generator.cpp [hunk 1]
```diff
@@ -86,15 +86,15 @@ void CalculateBestPedestrianSegments(string const & mwmPath, TCountryId const &
     return infoGetter->GetLimitRectForLeaf(c);
   };
 
-  CHECK_EQUAL(index.GetMwmId().GetInfo()->GetType(), MwmInfo::COUNTRY, ());
+  CHECK_EQUAL(dataSource.GetMwmId().GetInfo()->GetType(), MwmInfo::COUNTRY, ());
   auto numMwmIds = make_shared<NumMwmIds>();
   numMwmIds->RegisterFile(CountryFile(countryId));
 
   // Note. |indexRouter| is valid while |index| is valid.
```
Comment (mpimenov) [inline generator/transit_generator.cpp]: s/index/dataSource/

File: indexer/indexer_tests/index_builder_test.cpp [hunk 2]
```diff
@@ -52,11 +52,11 @@ UNIT_TEST(BuildIndexTest)
 
   {
     // Check that index actually works.
```
Comment (mpimenov) [inline indexer/indexer_tests/index_builder_test.cpp]: s/index/dataSource/

Comment (mpimenov) [reply indexer/indexer_tests/index_builder_test.cpp]: ... or not?

Comment (mpimenov) [reply indexer/indexer_tests/index_builder_test.cpp]: ok, leave it here as is

File: indexer/indexer_tests/index_test.cpp [hunk 3]
```diff
@@ -28,9 +28,9 @@ namespace
 class IndexTest : public MwmSet::Observer
 {
 public:
-  IndexTest() { TEST(m_index.AddObserver(*this), ()); }
+  IndexTest() { TEST(m_dataSource.AddObserver(*this), ()); }
```
Comment (mpimenov) [inline indexer/indexer_tests/index_test.cpp]: s/IndexTest/DataSourceTest/

also the filename

File: search/search_tests/house_detector_tests.cpp [hunk 4]
```diff
@@ -306,7 +306,7 @@ UNIT_TEST(HS_FindHouseSmoke)
     //m2::PointD pt = FindHouse(index, streetName, "28", 50);
```
Comment (tatiana-yan) [inline search/search_tests/house_detector_tests.cpp]: s/index/dataSource/g

File: routing/speed_camera.cpp [hunk 5]
```diff
@@ -55,7 +55,7 @@ uint8_t CheckCameraInPoint(m2::PointD const & point, DataSourceBase const & inde
       speedLimit = ReadCameraRestriction(ft);
   };
 
-  index.ForEachInRect(f,
+  dataSource.ForEachInRect(f,
```
Comment (mpimenov) [inline routing/speed_camera.cpp]: clang-format

File: search/engine.hpp [hunk 6]
```diff
@@ -90,7 +90,7 @@ class Engine
   };
 
   // Doesn't take ownership of index and categories.
```
Comment (mpimenov) [inline search/engine.hpp]: s/index/dataSource/

File: search/search_quality/assessment_tool/main_model.cpp [hunk 7]
```diff
@@ -30,7 +30,7 @@ using namespace std;
 // MainModel ---------------------------------------------------------------------------------------
 MainModel::MainModel(Framework & framework)
   : m_framework(framework)
-  , m_index(m_framework.GetIndex())
+  , m_index(m_framework.GetDataSource())
```
Comment (tatiana-yan) [inline search/search_quality/assessment_tool/main_model.cpp]: s/m_index/m_dataSource

File: search/search_integration_tests/processor_test.cpp [hunk 8]
```diff
@@ -702,7 +702,7 @@ UNIT_CLASS_TEST(ProcessorTest, TestCategories)
     TEST(ResultsMatch(request->Results(), rules), ());
     for (auto const & result : request->Results())
     {
-      EditableDataSource::FeaturesLoaderGuard loader(m_index, wonderlandId);
+      EditableDataSource::FeaturesLoaderGuard loader(m_dataSource, wonderlandId);
       FeatureType ft;
       TEST(loader.GetFeatureByIndex(result.GetFeatureID().m_index, ft), ());
```
Comment (tatiana-yan) [inline search/search_integration_tests/processor_test.cpp]: index

File: search/search_integration_tests/processor_test.cpp [hunk 9]
```diff
@@ -605,7 +605,7 @@ UNIT_CLASS_TEST(ProcessorTest, TestPostcodes)
     while (!features->GetBit(index))
       ++index;
 
-    EditableDataSource::FeaturesLoaderGuard loader(m_index, countryId);
+    EditableDataSource::FeaturesLoaderGuard loader(m_dataSource, countryId);
     FeatureType ft;
     TEST(loader.GetFeatureByIndex(::base::checked_cast<uint32_t>(index), ft), ());
```
Comment (tatiana-yan) [inline search/search_integration_tests/processor_test.cpp]: index

File: search/search_quality/assessment_tool/main_model.cpp (hunk 7 above)
Comment (mpimenov) [inline search/search_quality/assessment_tool/main_model.cpp]: s/m_index/m_dataSource/

Comment (tatiana-yan) [reply search/search_quality/assessment_tool/main_model.cpp]: и в search/search_quality/assessment_tool/main_model.hpp тоже

File: search/engine.hpp (hunk 6 above)
Comment (tatiana-yan) [inline search/engine.hpp]: s/index/dataSource

File: openlr/openlr_match_quality/openlr_assessment_tool/mainwindow.cpp [hunk 10]
```diff
@@ -124,7 +124,7 @@ class PointsControllerDelegate : public PointsControllerDelegateBase
 public:
   PointsControllerDelegate(Framework & framework)
     : m_framework(framework)
-    , m_index(framework.GetIndex())
+    , m_index(framework.GetDataSource())
```
Comment (tatiana-yan) [inline openlr/openlr_match_quality/openlr_assessment_tool/mainwindow.cpp]: s/m_index/m_dataSource
тут, в 232 строке и везде в этом классе

File: map/routing_manager.cpp [hunk 11]
```diff
@@ -364,7 +364,7 @@ RouterType RoutingManager::GetLastUsedRouter() const
 
 void RoutingManager::SetRouterImpl(RouterType type)
 {
-  auto const indexGetterFn = m_callbacks.m_indexGetter;
+  auto const indexGetterFn = m_callbacks.m_dataSourceGetter;
```
Comment (tatiana-yan) [inline map/routing_manager.cpp]: s/index/dataSource

File: map/routing_manager.cpp [hunk 12]
```diff
@@ -380,10 +380,10 @@ void RoutingManager::SetRouterImpl(RouterType type)
   auto numMwmIds = make_shared<NumMwmIds>();
   m_delegate.RegisterCountryFilesOnRoute(numMwmIds);
 
-  auto & index = m_callbacks.m_indexGetter();
+  auto & index = m_callbacks.m_dataSourceGetter();
```
Comment (tatiana-yan) [inline map/routing_manager.cpp]: s/index/dataSource

File: map/framework.cpp [hunk 13]
```diff
@@ -2914,7 +2914,7 @@ bool Framework::GetEditableMapObject(FeatureID const & fid, osm::EditableMapObje
   auto const & editor = osm::Editor::Instance();
   emo.SetEditableProperties(editor.GetEditableProperties(ft));
 
-  auto const & index = m_model.GetIndex();
+  auto const & index = m_model.GetDataSource();
```
Comment (tatiana-yan) [inline map/framework.cpp]: s/index/dataSource

File: map/framework.cpp [hunk 14]
```diff
@@ -2877,19 +2877,19 @@ bool Framework::CreateMapObject(m2::PointD const & mercator, uint32_t const feat
                                 osm::EditableMapObject & emo) const
 {
   emo = {};
-  auto const & index = m_model.GetIndex();
+  auto const & index = m_model.GetDataSource();
```
Comment (tatiana-yan) [inline map/framework.cpp]: s/index/dataSource

File: generator/utils.hpp [hunk 15]
```diff
@@ -25,17 +25,17 @@ class SingleMwmDataSource
...
 
 private:
-  DataSource m_index;
+  DataSource m_dataSource;
   platform::LocalCountryFile m_countryFile;
   MwmSet::MwmId m_mwmId;
 };
 
-void LoadIndex(DataSourceBase & index);
+void LoadDataSource(DataSourceBase & index);
```
Comment (tatiana-yan) [inline generator/utils.hpp]: s/index/dataSource

File: generator/utils.hpp [hunk 16]
```diff
@@ -25,17 +25,17 @@ class SingleMwmDataSource
   /// \param mwmPath is a path to mwm which should be registerd in Index.
```
Comment (tatiana-yan) [inline generator/utils.hpp]: s/Index/DataSource

File: generator/transit_generator.cpp (hunk 1 above)
Comment (syershov) [reply generator/transit_generator.cpp]: done

File: indexer/indexer_tests/index_test.cpp (hunk 3 above)
Comment (syershov) [reply indexer/indexer_tests/index_test.cpp]: done

File: search/search_tests/house_detector_tests.cpp (hunk 4 above)
Comment (syershov) [reply search/search_tests/house_detector_tests.cpp]: done

File: routing/speed_camera.cpp (hunk 5 above)
Comment (syershov) [reply routing/speed_camera.cpp]: done

File: search/engine.hpp (hunk 6 above)
Comment (syershov) [reply search/engine.hpp]: done

File: search/search_quality/assessment_tool/main_model.cpp (hunk 7 above)
Comment (syershov) [reply search/search_quality/assessment_tool/main_model.cpp]: done

Comment (syershov) [reply search/search_quality/assessment_tool/main_model.cpp]: done

File: search/engine.hpp (hunk 6 above)
Comment (syershov) [reply search/engine.hpp]: done

File: openlr/openlr_match_quality/openlr_assessment_tool/mainwindow.cpp (hunk 10 above)
Comment (syershov) [reply openlr/openlr_match_quality/openlr_assessment_tool/mainwindow.cpp]: done

File: map/routing_manager.cpp (hunk 11 above)
Comment (syershov) [reply map/routing_manager.cpp]: done

File: map/routing_manager.cpp (hunk 12 above)
Comment (syershov) [reply map/routing_manager.cpp]: done

File: map/framework.cpp (hunk 13 above)
Comment (syershov) [reply map/framework.cpp]: done

File: map/framework.cpp (hunk 14 above)
Comment (syershov) [reply map/framework.cpp]: done

File: generator/utils.hpp (hunk 15 above)
Comment (syershov) [reply generator/utils.hpp]: done

File: generator/utils.hpp (hunk 16 above)
Comment (syershov) [reply generator/utils.hpp]: done

**Comment by tatiana-yan** [comment]
.//track_analyzing/track_matcher.hpp:    void FillCandidatesWithNearbySegments(DataSourceBase const & index, routing::IndexGraph const & graph,
.//track_analyzing/track_matcher.cpp:    DataSourceBase const & index, IndexGraph const & graph, VehicleModelInterface const & vehicleModel,
.//feature_list/feature_list.cpp:  Processor(DataSourceBase const & index)
.//feature_list/feature_list.cpp:  DataSource index;
.//feature_list/feature_list.cpp:    DataSource::FeaturesLoaderGuard loader(index, mwmId);
.//search/search_quality/features_collector_tool/features_collector_tool.cpp:  DataSource index;
.//search/search_quality/search_quality_tool/search_quality_tool.cpp:int FindResult(DataSourceBase & index, string const & mwmName, uint32_t const featureId, double const lat,
.//search/search_quality/search_quality_tool/search_quality_tool.cpp:void CheckCompleteness(string const & path, m2::RectD const & viewport, DataSourceBase & index,
.//search/search_quality/search_quality_tool/search_quality_tool.cpp:  DataSource index;
.//map/booking_availability_filter.cpp:void PrepareData(DataSourceBase const & index, search::Results const & results,
.//openlr/openlr_stat/openlr_stat.cpp:void LoadIndexes(std::string const & pathToMWMFolder, std::vector<DataSource> & indexes)
.//openlr/openlr_stat/openlr_stat.cpp:  std::vector<DataSource> indexes(numThreads);
.//openlr/graph.cpp:Graph::Graph(DataSourceBase const & index, shared_ptr<CarModelFactory> carModelFactory)
.//openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.hpp:              DataSourceBase const & index,
.//openlr/graph.hpp:  Graph(DataSourceBase const & index, std::shared_ptr<routing::CarModelFactory> carModelFactory);
.//openlr/openlr_tests/decoded_path_test.cpp:void TestSerializeDeserialize(openlr::Path const & path, DataSource const & index)
.//openlr/openlr_tests/decoded_path_test.cpp:  DataSource index;
.//openlr/openlr_tests/decoded_path_test.cpp:  DataSource::FeaturesLoaderGuard const guard(index, regResult.first);
.//openlr/openlr_tests/decoded_path_test.cpp:  WithRoad(points, [&points](DataSource const & index, FeatureType & road) {
.//openlr/openlr_tests/decoded_path_test.cpp:  WithRoad({{0, 0}, {0, 1}, {1, 0}, {1, 1}}, [](DataSource const & index, FeatureType & road) {
.//android/jni/com/mapswithme/maps/Framework.cpp:DataSourceBase const & Framework::GetIndex() { return m_work.GetIndex(); }
.//android/jni/com/mapswithme/maps/Framework.hpp:    DataSourceBase const & GetIndex();


и ещё в iphone/ много всяких index, indexPath и т.п., с ними тоже надо разобраться

**Comment by tatiana-yan** [comment]
.//openlr/openlr_match_quality/openlr_assessment_tool/traffic_mode.hpp: class Index

**Comment by tatiana-yan** [comment]
.//track_analyzing/track_matcher.cpp:  index.ForEachInRect(

**Comment by syershov** [comment]
JTALL

**Comment by mpimenov** [comment]
Предлагаю написать в коммите, что переименовывается именно index в dataSource.

**Comment by syershov** [comment]
done

**Pull request merged**